src/components/AddTodoForm.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react'; 
+import PropTypes from 'prop-types';
+import InputWithLabel from './InputWithLabel';
+
+function AddTodoForm({ onAddTodo }) {
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const handleTitleChange = (event) => {
+    const newTodoTitle = event.target.value;
+    setTodoTitle(newTodoTitle);
+  };
+  function handleAddTodo(event) {
+    event.preventDefault();
+
+    setTodoTitle('');
+    onAddTodo(todoTitle);
```

Comment:
this probably isn't causing a bug, but only because `setState` operations are all technically asynchronous, so it probably hasn't changed to an empty string already when the value is passed to the handler. 

In general though it'd be safer to call the handler prop first, then the setter, so you have no chance of losing the value before you "capture" it: 

```suggestion
    onAddTodo(todoTitle);
    setTodoTitle('');
```